exporters/common/src/main/java/io/opentelemetry/exporter/internal/grpc/GrpcExporter.java

```diff
@@ -66,39 +68,17 @@ public CompletableResultCode export(T exportRequest, int numItems) {
         },
         (response, throwable) -> {
           exporterMetrics.addFailed(numItems);
+
+          logFailureMessage(response, throwable);
+
           switch (response.grpcStatusValue()) {
-            case GRPC_STATUS_UNIMPLEMENTED:
-              if (loggedUnimplemented.compareAndSet(false, true)) {
-                GrpcExporterUtil.logUnimplemented(
-                    internalLogger, type, response.grpcStatusDescription());
-              }
-              break;
-            case GRPC_STATUS_UNAVAILABLE:
-              logger.log(
-                  Level.SEVERE,
-                  "Failed to export "
-                      + type
-                      + "s. Server is UNAVAILABLE. "
-                      + "Make sure your collector is running and reachable from this network. "
-                      + "Full error message:"
-                      + response.grpcStatusDescription());
+            case GRPC_STATUS_UNKNOWN:
+              result.failExceptionally(FailedExportException.grpcFailedExceptionally(throwable));
               break;
             default:
-              logger.log(
-                  Level.WARNING,
-                  "Failed to export "
-                      + type
-                      + "s. Server responded with gRPC status code "
-                      + response.grpcStatusValue()
-                      + ". Error message: "
-                      + response.grpcStatusDescription());
+              result.failExceptionally(FailedExportException.grpcFailedWithResponse(response));
```

Comment:
The semantics behind when we call `grpcFailedExceptionally` and `grpcFailedWithResponse` are pretty loose right now - why is unknown status `grpcFailedExceptionally` while the other status codes are `grpcFailedWithResponse`? Makes you think that `FailedExporterException.GrpcExportException` can have non-null responses for both `getResponse()` and `getCause()`. But I don't think this is actually the case. I do think its one or the other.

 I think the fault lies with the GrpcSender interface rather than anything in this PR. Note that this same problem doesn't exist for HttpExporter? I think the HttpSender interface got it right. Opened #6658 to refactor the GrpcSender interface to match HttpExporter, which would clean up the semantics here.

No need to have this PR be blocked on merging that. I can solve merge conflicts and clean up #6658 after merging this.

Reply:
Yeah, I did feel this was a little odd but didn't want to delve too deep in this PR. Thanks for taking a look at fixing that.